update/Pos/Update/Poll/Logic/Base.hs

```diff
@@ -208,35 +210,39 @@ adoptBlockVersion winningBlk bv = do
 -- which currently adopted 'BlockVersion' can be applied.
 updateSlottingData
     :: (MonadError PollVerFailure m, MonadPoll m)
-    => EpochIndex -> m ()
-updateSlottingData epoch = do
-    sd@SlottingData {..} <- getSlottingData
+    => EpochIndex
+    -> m ()
+updateSlottingData epochIndex = do
     let errFmt =
             ("can't update slotting data, stored penult epoch is "%int%
              ", while given epoch is "%int%
              ")")
-    if | sdPenultEpoch + 1 == epoch -> updateSlottingDataDo sd
+
+    slottingData      <- getSlottingData
+    let nextEpochIndex = getNextEpochIndex slottingData
+    let nextEpochSD    = getNextEpochSlottingData slottingData
+
+    if | nextEpochIndex + 1 == epochIndex ->
```

Comment:
If I do that I get a test failure. 
We receive `EpochSlottingData` for `EpochIndex` that doesn't yet exist - like `nextEpochIndex + 1`, since we do have `nextEpochIndex` - that does make sense, right?